index.js

```diff
@@ -20,7 +20,6 @@ var createHistoryIndex = require('./lib/history-index.js')
 function Osm (opts) {
   if (!(this instanceof Osm)) return new Osm(opts)
   if (!opts.core) throw new Error('missing param "core"')
-  if (!opts.index) throw new Error('missing param "index"')
   if (!opts.storage) throw new Error('missing param "storage"')
```

Comment:
Maybe both `opts.index` and `opts.storage` should be set, or both should be falsey?